### ISSUE TITLE
Errors and partially successful response

### ISSUE THREAD

**Issue by aleksandrlat** [relay-tools/relay-hooks#176]
Hello @morrys!
Thank you for great library.

I'm trying to figure out if `useQuery` supports partial success and ability to access errors when partially successful response is returned.

I don't know if relay supports it or not. Or if it is possible to implement some workaround to access errors.
What I can see here https://github.com/relay-tools/relay-hooks/blob/master/src/FetchResolver.ts#L171
`next` method accepts entire payload including errors. What if we can set these errors and return from useQuery hook with partial success response?
I think this would be simple to implement, right?

**This is separate from above topic**

Also I found interesting idea in this comment https://github.com/facebook/relay/issues/1913#issuecomment-631652565
I'm curious if it would be possible to implement something in relay-hooks. Maybe with new hook?

I don't see official relay hooks support this.

**Comment by aleksandrlat** [comment]
Just something like this?
```
next: function (data) {
  var _a;

  if (data && data.errors) {
    error = data.errors
  }
  var store = environment.lookup(operation.fragment);
  promise = null;
  ((_a = operation.request.cacheConfig) === null || _a === void 0 ? void 0 : _a.poll) && updateLoading(false);
  resolveNetworkPromise_1();
  onNext(operation, store);
},
```

**Comment by aleksandrlat** [comment]
Or maybe `errors` or `serverErrors` can be returned from useQuery hook for backward compatibility with old code?

**Comment by morrys** [comment]
Hi @aleksandrlat,
I understand the need and your possible solution.

Have you tried to open an issue in Relay to discuss it with the relay team to understand their point of view?

**Comment by aleksandrlat** [comment]
@morrys I have not created issue in Relay. 
My colleague found this relatively old issue https://github.com/facebook/relay/issues/1913 in Relay repo

That is why I have not created new issue.

And also this old answer https://github.com/facebook/relay/issues/1816#issuecomment-304492071

Do you think I should open new issue?

All solutions from issues above are just workarounds. 
And if `useQuery` can return both it will not be workaround.

And what @hayes talks about in https://github.com/facebook/relay/issues/1913#issuecomment-631652565 would be even better

**Comment by aleksandrlat** [comment]
@morrys what do you think about my prev comment?

**Comment by morrys** [comment]
Hi @aleksandrlat,
sorry for the delay in replying but in this period I have many deadlines at work that keep me very busy :) 
I currently prefer to keep error handling at the network level as it could make the store inconsistent (https://github.com/morrys/react-relay-offline/issues/72)

**Comment by aleksandrlat** [comment]
Thank you @morrys!

But if I don't throw error in network layer (which I don't want to do) then store is "inconsistent" anyway.
I propose just returning error from the `useQuery` hook along with partially successful response. 

This does not affect store. It only allows to get error.

**Comment by aleksandrlat** [comment]
Hello @morrys!

Are you available for this issue? What do you think about my small proposal https://github.com/relay-tools/relay-hooks/issues/176#issuecomment-865405533? It does not affect store just allows to access error.

**Comment by morrys** [comment]
fixed with version [7.0.0](https://github.com/relay-tools/relay-hooks/releases/tag/v7.0.0)
